src/Amadeus/Client/Session/Handler/SoapHeader4.php

```diff
@@ -285,81 +285,84 @@ protected function createSoapHeaders($sessionData, $params, $messageName, $messa
             );
         }
 
-        //Send authentication info
-        if ($this->isAuthenticated === false) {
-            //Generate nonce, msg creation string & password digest:
-            $password = base64_decode($params->authParams->passwordData);
-            $creation = new \DateTime('now', new \DateTimeZone('UTC'));
-            $t = microtime(true);
-            $micro = sprintf("%03d", ($t - floor($t)) * 1000);
-            $creationString = $this->createDateTimeStringForAuth($creation, $micro);
-            $messageNonce = $this->generateUniqueNonce($params->authParams->nonceBase, $creationString);
-            $encodedNonce = base64_encode($messageNonce);
-            $digest = $this->generatePasswordDigest($password, $creationString, $messageNonce);
-
-            $securityHeaderXml = $this->generateSecurityHeaderRawXml(
-                $params->authParams->userId,
-                $encodedNonce,
-                $digest,
-                $creationString
-            );
+        // If not Security_Authenticate message - send authentication information in headers
+        if ($this->isNotSecurityAuthenticateMessage($messageName)) {
```

Comment:
I would prefer to have only one if stement (combine with the `$this->isAuthenticated === false` below).